client/keys/show.go

```diff
@@ -53,36 +51,43 @@ consisting of all the keys provided by name and multisig threshold.`,
 }
 
 func runShowCmd(cmd *cobra.Command, args []string) (err error) {
-	var info keyring.Info
+	k := new(keyring.Record)
 	clientCtx, err := client.GetClientQueryContext(cmd)
 	if err != nil {
 		return err
 	}
+	outputFormat := clientCtx.OutputFormat
 
 	if len(args) == 1 {
-		info, err = fetchKey(clientCtx.Keyring, args[0])
+		k, err = fetchKey(clientCtx.Keyring, args[0])
 		if err != nil {
 			return fmt.Errorf("%s is not a valid name or address: %v", args[0], err)
 		}
 	} else {
 		pks := make([]cryptotypes.PubKey, len(args))
 		for i, keyref := range args {
-			info, err := fetchKey(clientCtx.Keyring, keyref)
+			k, err := fetchKey(clientCtx.Keyring, keyref)
 			if err != nil {
 				return fmt.Errorf("%s is not a valid name or address: %v", keyref, err)
 			}
-
-			pks[i] = info.GetPubKey()
+			key, err := k.GetPubKey()
+			if err != nil {
+				return err
+			}
+			pks[i] = key
 		}
 
 		multisigThreshold, _ := cmd.Flags().GetInt(flagMultiSigThreshold)
-		err = validateMultisigThreshold(multisigThreshold, len(args))
-		if err != nil {
+
+		if err := validateMultisigThreshold(multisigThreshold, len(args)); err != nil {
 			return err
 		}
 
 		multikey := multisig.NewLegacyAminoPubKey(multisigThreshold, pks)
-		info, err = keyring.NewMultiInfo(defaultMultiSigKeyName, multikey)
+		emptyRecord := keyring.NewMultiRecord()
+		emptyRecordItem := keyring.NewMultiRecordItem(emptyRecord)
+
+		k, err = keyring.NewRecord(k.Name, multikey, emptyRecordItem)
```

Comment:
I see your're doing the `emptyRecord` hack multiple times, it seems weird to me. Does my comment above work? in this case we would do:
```suggestion
		k, err = keyring.NewMultiRecord(k.Name, multikey)
```

Reply:
Yeah. Comment above works.